ui/src/message_popup/mobile_prototype/texting_interface.jsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+
+import Divider from 'material-ui/Divider';
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
+
+import {StudentMessage, UserMessage, InfoMessage} from './message_components.jsx';
+
+export const TextBody = React.createClass({
+  propTypes: {
+    messages: React.PropTypes.array.isRequired
+  },
+  
+  render(){
+    const {messages} = this.props;
+    return (
+      <div>
+        {_.map(messages, (message) => { 
+          if(message.type === 'user'){
+            return <UserMessage text={message.text} key={message.key} />
+          }else if(message.type === 'student'){
+            return <StudentMessage text={message.text} key={message.key} />
+          }else{
+            return <InfoMessage text={message.text} key={message.key} />
+          }
+        })}
+      </div>
+    );
+  }
+});
+
+export const TextFooter = React.createClass({
+  propTypes: {
+    onQuestionDone: React.PropTypes.func.isRequired,
+    addResponse: React.PropTypes.func.isRequired,
+    isReadyToMoveOn: React.PropTypes.func.isRequired,
+    helpType: React.PropTypes.string.isRequired,
+    elapsedMs: React.PropTypes.number.isRequired,
+    nextButtonLabel: React.PropTypes.string.isRequired
+  },
+  
+  getInitialState(){
+    return ({
+      text: undefined,
+      initialMessageSent: false,
+    });
+  },
+  
+  onTextChange(event, value){
+    this.setState({text: value});
+  },
+  
+  onSendMessage(){
+    this.props.addResponse(this.state.text);
+    this.setState({initialMessageSent: true});
+  },
+  
+  onRevisionSendMessage(){
+    this.onSendMessage();
+  },
+  
+  onPassSendMessage(){
+    this.props.addResponse('');
+  },
+  
+  onNextQuestion(){
+    this.setState(this.getInitialState());
```

Comment:
In general this isn't a great pattern in React - it's better to update state and then let new components be created and destroyed based on what `render` returns rather than managing the state changes like this manually.  That's one of the main benefits, that you often don't have to think of resetting state.

Here, if we remove this what would happen?